src/de.rs

```diff
@@ -82,13 +84,57 @@ where
     Ok(ret)
 }
 
+/// Deserializes an IO stream into a type.
+///
+/// This function will attempt to read a TOML document from `r` and
+/// deserialize to `T` .
+///
+/// # Examples
+///
+/// ```rust
+/// #[macro_use]
+/// extern crate serde_derive;
+/// extern crate toml;
+///
+/// use std::fs::File;
+///
+/// #[derive(Debug, Deserialize)]
+/// struct Config {
+///     title: String,
+///     owner: Owner,
+/// }
+///
+/// #[derive(Debug, Deserialize)]
+/// struct Owner {
+///     name: String,
+/// }
+///
+/// fn main() {
+/// # }
+/// # fn fake_main() {
+///     let mut file = File::open("config.toml").unwrap();
+///     let config: Config = toml::from_reader(&mut file).unwrap();
+///
+///     println!("{:?}", config);
+/// }
+/// ```
+pub fn from_reader<R, T>(r: &mut R) -> Result<T, Error>
+where
+    R: io::Read,
+    T: de::DeserializeOwned,
+{
+    let mut buf = Vec::new();
+    r.read_to_end(&mut buf)?;
+    from_slice(&buf)
+}
+
 /// Errors that can occur when deserializing a type.
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug, Clone)]
```

Comment:
Unfortunately thtis removal is a breaking change, so we'll need to keep these traits.